build directory doesn't exist

```
ld.lld: error: cannot open output file build/sparser_dpu: No such file or directory
clang-9: error: ld.lld command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:27: dpu] Error 1
```

`HOST_TARGET` also appears on used in the makefile and could be removed.

--- a/sparser/.gitignore
+++ b/sparser/.gitignore
@@ -17,6 +17,7 @@ microbenchmarks/data
 _data/
 *.dSYM/
 bench
+build/
 out
 avro-c-1.8.2/build
 common/avro

--- a/sparser/sparser_dpu/Makefile
+++ b/sparser/sparser_dpu/Makefile
@@ -6,13 +6,12 @@ NR_DPUS ?= 2
 QUERY ?= aaba
 SIMULATOR ?= 1
 HOST_FLAGS := -DNR_TASKLETS=${NR_TASKLETS} -DNR_DPUS=${NR_DPUS} -DSIMULATOR=${SIMULATOR}
-#DPU_FLAGS:= -I/home/larry/workspace/upmem-2020.2.1-Linux-x86_64/include/dpu #-L/home/larry/workspace/upmem-2020.2.1-Linux-x86_64/lib -ldpu 
+#DPU_FLAGS:= -I/home/larry/workspace/upmem-2020.2.1-Linux-x86_64/include/dpu #-L/home/larry/workspace/upmem-2020.2.1-Linux-x86_64/lib -ldpu
 DPU_FLAGS:= `dpu-pkg-config --cflags --libs dpu`
 # dpu
 DPU_SOURCES := $(wildcard ${DPU_DIR}/*.c)
 
 BUILDDIR ?= build
-HOST_TARGET := ${BUILDDIR}/sparser_host
 DPU_TARGET := ${BUILDDIR}/sparser_dpu
 
 .PHONY: all all_dpu dpu debug asm clean
@@ -24,6 +23,7 @@ sparser:
 	cc -g $(INCLUDES) -O3 $(CFLAGS) -DNR_DPUS=${NR_DPUS} -DNR_TASKLETS=${NR_TASKLETS} -DQUERY=${QUERY} -c -o query_driver.o  query_driver.cpp
 
 dpu:
+	mkdir -p $(BUILDDIR)
 	dpu-upmem-dpurte-clang -Wall -Wextra -Werror -g -I ../dpu_common.h -I dpu/strstr_dpu.h -O -DNR_TASKLETS=${NR_TASKLETS} -o ${DPU_TARGET} dpu/strstr_dpu_nowrite.c
 
 dpu_host:

